refactor: remove v1 MetadataPopover component

pedesen opened the pull request:
## Description

:bulb: Please review the commits individually, because the first two commits are mostly deleting and moving files, which caused a big changeset.

- Remove v1 MetadataPopover component
- Move MetadataPopover/v2 -> MetadataPopover
- Explicitly pass data from MetadataPopover component to Details component, adjust tests
- Fix capitalization of instance count text

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

related to #33213

pedesen committed:
refactor: remove dep to flowNodeMetadataStore from MetadataPopover

pedesen committed:
fix: use correct capitalization of instance count text

pedesen commented on operate/client/src/App/ProcessInstance/TopPanel/MetadataPopover/index.tsx:
```diff
@@ -6,100 +6,290 @@
+        {instanceCount !== null && instanceCount > 1 && !elementInstanceKey && (
+          <>
+            <Header
+              title={`This Element instance triggered ${instanceCount} times`}
```
Good point, I fixed it.

pedesen commented on operate/client/src/App/ProcessInstance/TopPanel/MetadataPopover/types.ts:
```diff
@@ -119,8 +113,8 @@ function createV2InstanceMetadata(
-    state: (userTaskState ??
-      elementInstance.state) as V2InstanceMetadata['state'],
+    elementInstanceState: elementInstance.state,
+    userTaskState: userTaskState,
```
I decided to split the `state` field here.

pedesen commented on operate/client/src/App/ProcessInstance/TopPanel/MetadataPopover/types.ts:
```diff
@@ -129,10 +123,7 @@ function createV2InstanceMetadata(
     hasIncident: elementInstance.hasIncident,
     incidentKey: elementInstance.incidentKey,
     tenantId: elementInstance.tenantId,
-    flowNodeInstanceId: elementInstance.elementInstanceKey,
```
This already exists as `elementInstanceKey`

pedesen commented on operate/client/src/App/ProcessInstance/TopPanel/MetadataPopover/Details/index.tsx:
```diff
@@ -136,28 +130,40 @@ const Details: React.FC<Props> = ({metaData, flowNodeId}) => {
                 >
-                  {`${calledDecisionDefinitionName} - ${calledDecisionInstanceId}`}
+                  {`${calledDecisionInstance.decisionDefinitionName} - ${calledDecisionInstance.decisionEvaluationInstanceKey}`}
                 </Link>
```
the decision definition always exists along with a decision instance.

vsgoulart commented:
@pedesen Please take a look in [this PR](https://github.com/camunda/camunda/pull/40594)

I added this tool to identify dead code. After merging it we could rebase this branch and run it here to double check if we didn't miss anything

pedesen committed:
refactor: remove unused mock data

pedesen commented:
> @pedesen Please take a look in https://github.com/camunda/camunda/pull/40594
> 
> I added this tool to identify dead code. After merging it we could rebase this branch and run it here to double check if we didn't miss anything

@vsgoulart I ran it and found unused mock data which I removed.

vsgoulart commented on operate/client/src/App/ProcessInstance/TopPanel/MetadataPopover/types.ts:
```diff
@@ -17,13 +16,15 @@ import type {
 
-// V2 Element Instance Metadata - extends the old structure but with v2 element instance fields will be removed after other components migration
-type V2InstanceMetadata = {
+type TaskState = ElementInstance['state'] | UserTask['state'];
```
Why are we calculating the the `TaskState` like this?

pedesen replied:
Sorry, this is a leftover. Previously we were merging the two states to show them as "state" in the details popover. But I decided to split this up. We can remove the TaskState type and just use it like this:

```
  userTaskState?: UserTask['state'];
  elementInstanceState: ElementInstance['state'];
```

vsgoulart commented:
@pedesen Maybe you can solve this in this PR too but we have a function on the `types.ts` file [here](https://github.com/camunda/camunda/blob/9a7b056f2e3bb36a2226f286eb92284ea9a9b59e/operate/client/src/App/ProcessInstance/TopPanel/MetadataPopover/types.ts#L69C10-L69C32)...

pedesen commented:
> @pedesen Maybe you can solve this in this PR too but we have a function on the `types.ts` file [here](https://github.com/camunda/camunda/blob/9a7b056f2e3bb36a2226f286eb92284ea9a9b59e/operate/client/src/App/ProcessInstance/TopPanel/MetadataPopover/types.ts#L69C10-L69C32)...

Let me refactor this in a followup PR, where I also move some queries from MetadataPopover to DetailsModal.

pedesen committed:
fix: fix user task type
